posthog/hogql/test/test_query.py

```diff
@@ -1446,7 +1446,7 @@ def test_hogql_query_filters(self):
                 )
             query = "SELECT event, distinct_id from events WHERE distinct_id={distinct_id} and {filters}"
             filters = HogQLFilters(
-                properties=[EventPropertyFilter(key="index", operator="exact", value=4, type="event")]
+                properties=[EventPropertyFilter(key="index", operator="exact", value="4", type="event")]
```

Comment:
This was erroring with a ClickHouse exception, due to different behaviour in Pydantic v1 and v2.

### ClickHouse Error

```
DB::Exception: There is no supertype for types String, Float64 because some of them are String/FixedString and some of them are not: while executing 'FUNCTION equals(replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(properties, 'index'), ''), 'null'), '^"|"$', '') : 17, 4. : 10) -> equals(replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(properties, 'index'), ''), 'null'), '^"|"$', ''), 4.) Nullable(UInt8) : 18'. Stack trace:9
```

### Pydantic

See the populated value property (right at the end).

#### Pydantic v1
```python
EventPropertyFilter(key="index", operator="exact", value=4, type="event")
> EventPropertyFilter(key='index', label=None, operator=<PropertyOperator.exact: 'exact'>, type='event', value='4')
```

#### Pydantic v2

```python
EventPropertyFilter(key="index", operator="exact", value=4, type="event")
> EventPropertyFilter(key='index', label=None, operator=<PropertyOperator.exact: 'exact'>, type='event', value=4.0)
```